### PR TITLE
(PUP-11321) Allow loading of safe classes from the persistence store

### PR DESCRIPTION
**Pull request by joshcooper**
Revert changes introduced for PUP-10820 and fix binary values in the persistence store a different way

### CODE REVIEW

**Comment by joshcooper** [comment]
This is blocked on https://github.com/puppetlabs/puppet/pull/8829

**Comment by joshcooper** [comment]
@ciprianbadescu when you get a chance, could you merge the ruby-openssl fix & rebase this PR on 6.x?

**Review by ciprianbadescu — APPROVED** [review]

File: lib/puppet/transaction/persistence.rb [hunk 1]
```diff
@@ -6,6 +6,26 @@
...
+      # URI is excluded, because it serializes all instance variables including the
+      # URI parser. Better to serialize the URL encoded representation.
+      SemanticPuppet::Version,
+      # SemanticPuppet::VersionRange has many nested classes and is unlikely to be
+      # used directly, so ignore it
+      Puppet::Pops::Time::Timestamp,
+      Puppet::Pops::Time::TimeData,
+      Puppet::Pops::Time::Timespan,
+      Puppet::Pops::Types::PBinaryType::Binary,
+      # Puppet::Pops::Types::PSensitiveType::Sensitive values are excluded from
```
Comment (alexjfisher) [inline lib/puppet/transaction/persistence.rb:24]: @joshcooper Are they definitely?  I've managed to hit
```
Error: Transaction store file /opt/puppetlabs/puppet/cache/state/transactionstore.yaml is corrupt ((/opt/puppetlabs/puppet/cache/state/transactionstore.yaml): Tried to load unspecified class: Puppet::Pops::Types::PSensitiveType::Sensitive); replacing
```
in Puppet 6.28.0

Comment (joshcooper) [reply lib/puppet/transaction/persistence.rb:24]: @alexjfisher oh interesting... are you able to reproduce? Or was it a one-time thing (for example after upgrading from an older version to 6.28.0)?

**Pull request merged**